plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1290,6 +1257,243 @@ public void testJoinPushdown()
         }
     }
 
+    @DataProvider
+    public Object[][] joinOperators()
+    {
+        if (hasBehavior(SUPPORTS_JOIN_PUSHDOWN_WITH_FULL_JOIN)) {
+            return new Object[][] {{"JOIN"}, {"LEFT JOIN"}, {"RIGHT JOIN"}, {"FULL JOIN"}};
+        }
+        else {
+            return new Object[][] {{"JOIN"}, {"LEFT JOIN"}, {"RIGHT JOIN"}};
+        }
+    }
+
+    @Test
+    public void testJoinPushdownInequalityStrictNumeric()
```

Comment:
seems parts of testJoinPushdownInequalityStrictVarchar and testJoinPushdownInequalityStrictNumeric could be reused with different Supplier/Function. Will it make sense?

Reply:
to be honest didn't get it, which part you think we can put to supplier?

Reply:
expectJoinPushdown vs expectVarcharJoinPushdown

Reply:
<img width="1472" alt="image" src="https://user-images.githubusercontent.com/3051719/199787101-5c90dfa6-7855-442a-b022-d9149a2aabfc.png">

Reply:
and expected plan is different for LEFT join

Reply:
I think it could be extracted in case if we can create some common plan for NotFully pushdowns,
but look like the common one: 
```
PlanMatchPattern joinOverTableScans =
                node(JoinNode.class,
                        anyTree(node(TableScanNode.class)),
                        anyTree(node(TableScanNode.class)));
```

is not matched properly if we have any exchanges in tree :(

Reply:
After our offline discussion, I introduced specific join assertions, so now this complexity is removed and we don't need separate test methods for inequality operators